packages/vite/src/node/plugins/css.ts

```diff
@@ -2058,6 +2058,10 @@ async function doUrlReplace(
   }
 
   let newUrl = await replacer(rawUrl)
+  if (rawUrl === newUrl) {
+    return matched
+  }
```

Comment:
Maybe we could add a comment here that says that the replacer will bail out when checking the first variable in the case of a concat and this is why it works? I think it would be better to check this in `skipUrlReplacer` but I understand it is a big refactoring to pass the `variablePrefix` to it. So good to go for me if we add the comment to avoid confusion later on.

Reply:
That would be better. Added a comment for now 👍

Reply:
`rawUrl` can originally have a space and be quoted, and we are removing the quotes here in that case https://github.com/vitejs/vite/pull/20068/files#diff-2cfbd4f4d8c32727cd8e1a561cffbde0b384a3ce0789340440e144f9d64c10f6R2053

Or am I reading this wrong? I think it is probably good to have a local `unquotedUrl` instead of overriding `rawUrl`, as this is indeed confusing.

Reply:
Ah, I overlooked that 🤦

Reply:
Let me come up with a more proper way to fix this.

Reply:
I've changed the impl to allow returning `false` from `CssUrlReplacer`. The previous impl didn't handle interpolation properly and that is not possible to have it inside `skipUrlReplacer` because the behavior is different for each preprocessors.

Reply:
I see now that the comment confused me as it was
```
// If the replacer output is same with the input, we don't need to wrap it with quotes.
```
but the code was returning `matched`, so it was working before but the new code is way better and you fixed other issues so this was good in the end :)